lib/fields.nix

```diff
@@ -0,0 +1,33 @@
+{ lib }:
+{
+  /**
+    Creates a Field attribute set. mkField accepts an attribute set with the following keys:
+
+     Example:
+       mkField { }  // => { _type = "field"; }
+       mkField { default = "foo"; } // => { _type = "field"; default = "foo"; }
+       mkField { optional = true; } // => { _type = "field"; optional = true; }
+  */
+  mkField =
+    {
+      # Default value used when no definition is given in the configuration.
+      default ? null,
+      # Textual representation of the default, for the manual.
+      defaultText ? null,
+      # Example value used in the manual.
+      example ? null,
+      # String describing the field.
+      description ? null,
+      # Related packages used in the manual (see `genRelatedPackages` in ../nixos/lib/make-fields-doc/default.nix).
+      relatedPackages ? null,
+      # Option type, providing type-checking and value merging.
+      type ? null,
+      # Whether the field is for NixOS developers only.
+      internal ? null,
```

Comment:
```suggestion
```

I don't think we've ever made us of the difference between `internal` and `visible`, and `visible` is the more capable one, supporting `"shallow"`.